### ISSUE TITLE
PSD: Some images that were read with 3.8.2 are not read with 3.9.4.

### ISSUE THREAD

**Issue by tc-wleite** [haraldk/TwelveMonkeys#738]
**Describe the bug**
I found two (similar) images that were read successfully with 3.8.2 PSD ImageIO plugin, but failed with an exception when using 3.9.4. 
It is important to notice that both images seem to be incomplete or corrupted, so this is actually not a bug. 

**Version information**
1. The version of the TwelveMonkeys ImageIO library in use.
3.8.2 (reads the images) and 3.9.4 (fails to read).

2. The *exact* output of `java --version`: 
openjdk version "11.0.13" 2021-10-19 LTS
OpenJDK Runtime Environment (build 11.0.13+8-LTS)
OpenJDK 64-Bit Server VM (build 11.0.13+8-LTS, mixed mode, sharing)

3. Extra information about OS version, server version, standalone program or web application packaging, executable wrapper, etc. Please state exact version numbers where applicable.
I tested in Windows 10 and 11.
While reading ~3K PSD images from different sources, types and resolutions, I found this issue. I tested later with a simple standalone program (code below) to confirm the behavior.

**To Reproduce**
A simple read with a standalone program below, using the PSD ImageIO plugin.

**Expected behavior**
It would be nice if 3.9.4 can also read these images, as 3.8.2 does, but as mentioned before, maybe it is not possible/desirable, as they seem to be incomplete/corrupted PSDs. 

**Example code**
```java
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;

import javax.imageio.ImageIO;

public class PSDReadTest {
    public static void main(String[] args) throws IOException {
        File input = new File("1.psd"); // Input image
        File output = new File("1.png"); // Output image

        BufferedImage image = ImageIO.read(input);
        ImageIO.write(image, "png", output);
    }
}
``` 

**Sample file(s)**
[1.zip](https://github.com/haraldk/TwelveMonkeys/files/10979067/1.zip)
[2.zip](https://github.com/haraldk/TwelveMonkeys/files/10979068/2.zip)
I had to compress these files with 7-ZIP, otherwise (with Windows integrated ZIP creation) they would exceed GitHub's 25 MB limit.
Please use it to expand them.

**Stack trace**
Running with the test program using 3.9.4, I get the following exception:
```text
Exception in thread "main" java.io.EOFException: Unexpected end of PackBits stream
	at com.twelvemonkeys.io.enc.PackBitsDecoder.readFully(PackBitsDecoder.java:186)
	at com.twelvemonkeys.io.enc.PackBitsDecoder.decode(PackBitsDecoder.java:143)
	at com.twelvemonkeys.io.enc.DecoderStream.fill(DecoderStream.java:179)
	at com.twelvemonkeys.io.enc.DecoderStream.read(DecoderStream.java:123)
	at java.base/java.io.SequenceInputStream.read(SequenceInputStream.java:199)
	at com.twelvemonkeys.imageio.stream.DirectImageInputStream.read(DirectImageInputStream.java:73)
	at java.desktop/javax.imageio.stream.ImageInputStreamImpl.readFully(ImageInputStreamImpl.java:351)
	at com.twelvemonkeys.imageio.plugins.psd.PSDImageReader.read8bitChannel(PSDImageReader.java:656)
	at com.twelvemonkeys.imageio.plugins.psd.PSDImageReader.readImageData(PSDImageReader.java:480)
	at com.twelvemonkeys.imageio.plugins.psd.PSDImageReader.read(PSDImageReader.java:429)
	at java.desktop/javax.imageio.ImageIO.read(ImageIO.java:1468)
	at java.desktop/javax.imageio.ImageIO.read(ImageIO.java:1315)
	at PSDReadTest.main(PSDReadTest.java:12)
```

**Screenshots**
These are the images produced by 3.8.2 (downscaled, to reduce file size):

1.psd
<img src="https://user-images.githubusercontent.com/7217001/225353945-8c180fd3-d049-4250-893d-6fc5ca5e32fb.jpg" width="256">

2.psd
<img src="https://user-images.githubusercontent.com/7217001/225353963-02434e44-c014-4d78-9fe9-6d132407143f.jpg" width="256">

**Additional context**
Only the second sample is read by ImageMagick, producing a very similar output that 3.8.2 does.

**Comment by haraldk** [comment]
Thanks! 

With your sample files I discovered a small issue introduced in the 3.9.x branches, for PackBits compression (the normal PSD compression). It would surface if the compressed runs are longer than the compressed data for each scan line. 

As you say, this is probably some kind of corruption in the data. Could also be a malfunctioning encoder. I wonder if Photoshop itself is able to open these files?

Anyway, I have a fix ready that will restore the 3.8.x behavior. It comes with a very small performance penalty but I don't think it's noticeable in practice.

**Comment by tc-wleite** [comment]
That is great! Thank you!

**Comment by haraldk** [comment]
Fixed in 606fd53823623ba866fc4e31ce7fa751278fd51d, feel free to test the latest snapshot build and report back. 😀

**Comment by tc-wleite** [comment]
> Fixed in [606fd53](https://github.com/haraldk/TwelveMonkeys/commit/606fd53823623ba866fc4e31ce7fa751278fd51d), feel free to test the latest snapshot build and report back. 😀

Hi @haraldk!
I tested with the latest snapshot and it worked great!

I collected a new set of PSD images and compared the results between 3.8.2 and the latest version.
I didn't found any PSDs read by 3.8.2 and not read by the newest version.

Thank you again for the quick response!
